src/SelfServe/SelfServeUtils.tsx

```diff
@@ -0,0 +1,191 @@
+import "reflect-metadata";
+import {
+  DropdownItem,
+  Node,
+  Info,
+  InputTypeValue,
+  Descriptor,
+  AnyInput,
+  NumberInput,
+  StringInput,
+  BooleanInput,
+  DropdownInput,
+  InputType
+} from "../Explorer/Controls/SmartUi/SmartUiComponent";
+
+export enum SelfServeTypes {
```

Comment:
This name is enum is really confusing to me. Am I understanding right that its defining different examples of using self serve? Can it be renamed to something more descriptive? The "Types" suffix makes me thing its related to the type system but it doesn't appear to be

Reply:
So this is what i m trying to achieve
when selfServeType is undefined -> show a loading spinner
when no selfserveType is passed from portal , I set the type to "none" -> this loads the data explorer in the normal path
when some selfServeType is set as a feature flag, but it is not supported on data explorer, I set it to "invaid" -> this shows "Invalid Self Serve type"